x-pack/plugins/lens/public/shared_components/drag_drop_bucket/buckets.tsx

```diff
@@ -5,162 +5,110 @@
-  'data-test-subj'?: string;
-  isDisabled?: boolean;
-  className?: string;
-}) => (
-  <EuiButtonEmpty
-    data-test-subj={dataTestSubj ?? 'lns-newBucket-add'}
-    size="xs"
-    iconType="plusInCircle"
-    onClick={onClick}
-    isDisabled={isDisabled}
-    flush="left"
-    className={className}
-  >
-    {label}
-  </EuiButtonEmpty>
-);
-
-interface BucketContainerProps {
-  isInvalid?: boolean;
-  invalidMessage: string;
-  onRemoveClick: () => void;
-  removeTitle: string;
-  isNotRemovable?: boolean;
-  children: React.ReactNode;
-  dataTestSubj?: string;
-}
-
-const BucketContainer = ({
-  isInvalid,
-  invalidMessage,
-  onRemoveClick,
-  removeTitle,
-  children,
-  dataTestSubj,
-  isNotRemovable,
-}: BucketContainerProps) => {
-  return (
-    <EuiPanel paddingSize="none" data-test-subj={dataTestSubj} hasShadow={false} hasBorder>
-      <EuiFlexGroup gutterSize="s" alignItems="center" responsive={false}>
-        <EuiFlexItem grow={false}>{/* Empty for spacing */}</EuiFlexItem>
-        <EuiFlexItem grow={false}>
-          <EuiIcon
-            size="s"
-            color={isInvalid ? 'danger' : 'subdued'}
-            type={isInvalid ? 'alert' : 'grab'}
-            title={
-              isInvalid
-                ? invalidMessage
-                : i18n.translate('xpack.lens.customBucketContainer.dragToReorder', {
-                    defaultMessage: 'Drag to reorder',
-                  })
-            }
-          />
-        </EuiFlexItem>
-        <EuiFlexItem grow={true}>{children}</EuiFlexItem>
-        <EuiFlexItem grow={false}>
-          <EuiButtonIcon
-            iconSize="s"
-            iconType="cross"
-            color="danger"
-            data-test-subj="lns-customBucketContainer-remove"
-            onClick={onRemoveClick}
-            aria-label={removeTitle}
-            title={removeTitle}
-            disabled={isNotRemovable}
-          />
-        </EuiFlexItem>
-      </EuiFlexGroup>
-    </EuiPanel>
-  );
-};
+import { DefaultBucketContainer } from './default_bucket_container';
+import type { BucketContainerProps } from './types';
 
 export const DraggableBucketContainer = ({
   id,
-  idx,
   children,
+  isInsidePanel,
+  Container = DefaultBucketContainer,
   ...bucketContainerProps
-}: {
-  id: string;
-  idx: number;
-  children: React.ReactNode;
-} & BucketContainerProps) => {
+}: Assign<
+  Omit<BucketContainerProps, 'draggableProvided'>,
+  {
+    id: string;
+    children: React.ReactNode;
+    isInsidePanel?: boolean;
+    Container?: React.FunctionComponent<BucketContainerProps>;
+  }
+>) => {
+  const { euiTheme } = useEuiTheme();
+
   return (
     <EuiDraggable
-      style={{ marginBottom: 4 }}
-      spacing="none"
-      index={idx}
       draggableId={id}
+      customDragHandle={true}
+      index={bucketContainerProps.idx}
+      isDragDisabled={bucketContainerProps.isNotDraggable}
+      style={!isInsidePanel ? { marginBottom: euiTheme.size.xs } : {}}
+      spacing="none"
```

Comment:
As only the drag handle now drags the item, we need to add the `hasInteractiveChildren` prop to `EuiDraggable` to fix the current `cursor`, `role`, and `tabindex` issues.

```suggestion
      spacing="none"
      hasInteractiveChildren
```